gobblin-hive-registration/src/main/java/org/apache/gobblin/hive/metastore/HiveMetaStoreBasedRegister.java

```diff
@@ -492,7 +499,7 @@ public void addOrAlterPartition(HiveTable table, HivePartition partition) throws
     }
   }
 
-  private void addOrAlterPartition(IMetaStoreClient client, Table table, HivePartition partition)
+  private void addOrAlterPartitionWithPushMode(IMetaStoreClient client, Table table, HivePartition partition)
```

Comment:
Not a blocker for this PR. But if "pull" mode is the way to go, maybe we should remove the "push" mode in future. Same for skipping diff computations on partitions. Not sure if it is needed.

Reply:
Since when we call add_partition, they do a check before really adding to the meta store, so the mainly cost here is about handling exception. In batch mode, most of the partitions it tries to register are not existed. In that case, I think push mode is a better way to go.

Reply:
Fair point!